did/resolver.go

```diff
@@ -64,3 +65,23 @@ func GetMethodForDID(did string) (Method, error) {
 	}
 	return Method(split[1]), nil
 }
+
+// ParseDIDResolution attempts to parse a DID Resolution Result or a DID Document
+func ParseDIDResolution(resolvedDID []byte) (*DIDResolutionResult, error) {
+	// first try to parse as a DID Resolution Result
+	var result DIDResolutionResult
```

Comment:
Note that `DIDResolutionResult` does not have `json` tags in the struct. I had to add them in https://github.com/TBD54566975/ssi-service/blob/4936f10da0bae4b1606f5b2362c32bb6eb02e170/pkg/service/did/resolve/universal.go#L55

Reply:
added